### PR TITLE
[FLINK-34639] Support debezium deserializer in OceanBase source connector

### PR DESCRIPTION
**Pull request by whhe**

### CODE REVIEW

**Comment by whhe** [comment]
This is a part of the incremental source implementation. PTAL @leonardBang @ruanhang1993

**Review by PatrickRen — COMMENTED** [review]
@whhe Thanks for the PR! I'm not familiar with OceanBase so I just left some general comments.

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-oceanbase-cdc/src/main/java/org/apache/flink/cdc/connectors/oceanbase/OceanBaseSource.java [hunk 1]
```diff
@@ -151,7 +152,7 @@ public Builder<T> logProxyHost(String logProxyHost) {
             return this;
         }
 
-        public Builder<T> logProxyPort(int logProxyPort) {
+        public Builder<T> logProxyPort(Integer logProxyPort) {
```
[the remark this answers is not in the document]
Comment (whhe) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-oceanbase-cdc/src/main/java/org/apache/flink/cdc/connectors/oceanbase/OceanBaseSource.java:154]: In this patch I also introduced the StartupMode from flink-cdc-base, and logProxyPort can be null when starting  connector with snapshot only mode.

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-oceanbase-cdc/pom.xml [hunk 2]
```diff
@@ -24,8 +24,9 @@ limitations under the License.
         <version>${revision}</version>
     </parent>
     <properties>
+        <oblogclient.version>1.1.1</oblogclient.version>
         <!-- Because of oceanbase docker image can not expose port quickly, so we need to specify testcontainers version to 1.15.3 -->
-        <jdbc.version>1.15.3</jdbc.version>
+        <testcontainers.version>1.15.3</testcontainers.version>
```
[the remark this answers is not in the document]
Comment (whhe) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-oceanbase-cdc/pom.xml:29]: The latest testcontainers once caused container startup failures, but I can have a try.

**Review by PatrickRen — APPROVED** [review]
@whhe Thanks for the update! LGTM

**Pull request merged**